### PR TITLE
Add user info tooltip to actor fields

### PR DESCRIPTION
**Pull request by cconard96**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !23761

### CODE REVIEW

**Review by trasher — APPROVED** [review]

**Comment by trasher** [comment]
You have CS to fix ;)

File: templates/components/itilobject/actors/field.html.twig [hunk 1]
```diff
@@ -54,16 +54,37 @@
    <select class="form-select" multiple="true" id="actor_{{ rand }}"
         {{ required ? 'required' : '' }}>
    {% for actor in actors %}
+      {% set unique_id = "user_opt_" ~ actortype ~ actor.itemtype ~ actor.items_id %}
       <option selected="true" value="{{ actor['itemtype'] ~ '_' ~ actor['items_id'] }}"
             data-itemtype="{{ actor['itemtype'] }}" data-items-id="{{ actor['items_id'] }}"
             data-use-notification="{{ actor['use_notification'] }}"
             data-alternative-email="{{ actor['alternative_email'] }}"
-            data-text="{{ actor['text']|verbatim_value }}" data-title="{{ actor['title']|verbatim_value }}">
+            data-text="{{ actor['text']|verbatim_value }}" data-title="{{ actor['title']|verbatim_value }}" data-glpi-popover-source="content{{ unique_id }}">
```
Comment (cedric-anne) [inline templates/components/itilobject/actors/field.html.twig:62]: Maybe this `data-glpi-popover-source` attribute should be placed only if popover is available.
To do so, you could probably generate popovers, store them into a variable, and print them after the select element.

```
{% set actors_popovers = [] %}
{% for actor in actors %}
    ....
    {% set popover %}
        popover code
    {% endset %}
    {% set actors_popovers = actors_popovers|merge([popover]) %}
{% endfor %}

{{ actors_popovers|join|raw }}
```

**Comment by orthagh** [comment]
Another info : this is related to !23761

**Comment by cconard96** [comment]
>     * most important one : hovering the popover should prevent hiding it (to be able to click on the links present in content). If not possible a longer delay for hide event could help.

I've not been able to get this to work at all. As far as I can tell, you have to change the trigger of the popover to be "manual" but in that mode, I've not been able to get the popover to show using ".popover('show')" on the element itself (The one that the popover is delegated to using the selector option) during the mouseenter event or the element that the popover was initialized on.
The popover initialization has to be delegated since we have to wait an unknown amount of time for the select2 library to initialize its dropdowns. It is the reason the qtip-based tooltips couldn't be used directly.

**Comment by cconard96** [comment]
All points addressed except keeping the popover open when hovering (added delay instead).

**Comment by orthagh** [comment]
Thank you, wil lcheck tomorrow

**Review by cedric-anne — COMMENTED** [review]
Seems OK to me, but I was not able to fix tests. Seems that generated HTML is not correctly parsed by `symfony/dom-crawler` component. I tried to remove all `</div>` added by this PR in javascripts, and tests are passing. As generated HTML contains about 10k lines, it is hard to know if it is malformed or if it is a bug in the crawler.

**Comment by orthagh** [comment]
> All points addressed except keeping the popover open when hovering (added delay instead).

OK. I think we could increase a little more the delay.
I just wake up and i'm not able to target the links :) . 1.5s or 2s should be fine

**Comment by cconard96** [comment]
Done

**Review by orthagh — APPROVED** [review]

**Comment by trasher** [comment]
Test sare failing, and you have some cs to fix ;)

**Comment by cconard96** [comment]
> Seems OK to me, but I was not able to fix tests. Seems that generated HTML is not correctly parsed by `symfony/dom-crawler` component. I tried to remove all `</div>` added by this PR in javascripts, and tests are passing. As generated HTML contains about 10k lines, it is hard to know if it is malformed or if it is a bug in the crawler.

Seems like an issue with ext-dom of its dependency libxml. It doesn't handle HTML inside script tags well at all.
https://github.com/symfony/symfony/issues/14542

It sounds like the DomCrawler component is not reliable unless working with pure HTML or JavaScript with no HTML tags that can be parsed. It doesn't take being in a JS context into account. By breaking apart some tags by using concatenation like "'<' + 'div>", it allows the DomCrawler/DOM extension to parse it correctly.

Either these tests should be redone with an End-to-End testing tool like Cypress, or inline scripts that can have HTML should be moved to separate JS files (should be done anyways).

**Comment by cconard96** [comment]
For now, I broke up the HTML tag strings.
They will be fixed properly when I move the code out of the twig template (another PR).

**Review by orthagh — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Pull request merged**